File: iPad/VerBank-CSTSv3-IOSClientApi/ClientStation/api/info/InfoOperator_TRADESERV1006.m
//
//  InfoOperator_TRADESERV1006.m
//  VerBank-IOSClientStation
//
//  Created by Allone on 15/8/31.
//  Copyright (c) 2015年 zhanglei. All rights reserved.
//

#import "InfoOperator_TRADESERV1006.h"

#import "Info_TRADESERV1006.h"
#import "APIDoc.h"
#import "MTP4CommDataInterface.h"
#import "API_IDEventCaptain.h"
#import "DocOperator.h"

@implementation InfoOperator_TRADESERV1006

- (void) onInfo:(InfoFather *)info {
    Info_TRADESERV1006 *tinfo = (Info_TRADESERV1006 *)info;
    
    [[[APIDoc getSystemDocCaptain] systemConfig] setTradeDay:[tinfo getTradeDay]];
    [[[APIDoc getSystemDocCaptain] systemConfig] setCloseStatus:[tinfo getCloseState]];
    
    if ([tinfo getCloseState] == CLOSESTATUS_CLOSE) {
        [API_IDEventCaptain fireEventChanged:OTHER_ON_SYSTEMCLOSE eventData:nil];
        NSArray *asArray = [[APIDoc getUserDocCaptain] getCDS_AccountStoreArray];
        for (CDS_AccountStore *as in asArray) {
            [[DocOperator getInstance] loadAccountData:[as getAccountID]];
        }
    } else {
        [API_IDEventCaptain fireEventChanged:OTHER_ON_SYSTEMOPEN eventData:nil];
        NSArray *asArray = [[APIDoc getUserDocCaptain] getCDS_AccountStoreArray];
        for (CDS_AccountStore *as in asArray) {
            [[DocOperator getInstance] loadAccountData:[as getAccountID]];
        }
        [API_IDEventCaptain fireEventChanged:DATA_ON_Rollover_Changed eventData:nil];
    }
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/info/InfoOperator_TRADESERV1006.java
package jedi.ex01.client.station.api.info;

import jedi.ex01.CSTS3.comm.info.InfoFather;
import jedi.ex01.CSTS3.comm.info.Info_TRADESERV1006;
import jedi.ex01.CSTS3.proxy.comm.AccountStore;
import jedi.ex01.CSTS3.proxy.comm.MTP4CommDataInterface;
import jedi.ex01.client.station.api.doc.DataDoc;
import jedi.ex01.client.station.api.doc.operator.DocOperator;
import jedi.ex01.client.station.api.event.API_IDEventCaptain;
import jedi.ex01.client.station.api.event.API_IDEvent_NameInterface;

public class InfoOperator_TRADESERV1006 extends InfoOperator {

	public void onInfo(InfoFather _info) {
		Info_TRADESERV1006 info = (Info_TRADESERV1006) _info;
		DataDoc.getInstance().getSystemConfig().setTradeDay(info.getTradeDay());
		DataDoc.getInstance().getSystemConfig().setCloseStatus(info.getCloseState());
		
		if (info.getCloseState() == MTP4CommDataInterface.CLOSESTATUS_CLOSE) {
			API_IDEventCaptain.fireEventChanged(API_IDEvent_NameInterface.OTHER_ON_SYSTEMCLOSE);
			AccountStore accoutStore = DataDoc.getInstance().getAccountStore();
			DocOperator.getInstance().loadAccountData(accoutStore.getAccountID());
		} else {
			API_IDEventCaptain.fireEventChanged(API_IDEvent_NameInterface.OTHER_ON_SYSTEMOPEN);
			API_IDEventCaptain.fireEventChanged(API_IDEvent_NameInterface.DATA_ON_Rollover_Changed);
		}
	}

}
